pal/src/thread/thread.cpp

```diff
@@ -2820,3 +2820,32 @@ void GetCurrentThreadStackLimits(ULONG_PTR* lowLimit, ULONG_PTR* highLimit)
     *highLimit = (ULONG_PTR) stackbase;
 #endif
 }
+
+static thread_local ULONG_PTR s_cachedThreadHighLimit = 0;
+
+bool IsAddressOnStack(ULONG_PTR address)
```

Comment:
Should it be better to check if there is an upper limit in case of `ulimit -s unlimited` ? `getrlimit(RLIMIT_STACK..`